### ISSUE TITLE
Python API does not implement read backtest log

### ISSUE THREAD

**Issue by klk-** [QuantConnect/Lean#5279]
#### Expected Behavior
Can use python API to download log for backtest

#### Actual Behavior
Not implemented currently

#### Potential Solution
Add read_backtest_log method to api

#### Reproducing the Problem
N/A

#### System Information
N/A

#### Checklist
- [ ] I have completely filled out this template
- [ ] I have confirmed that this issue exists on the current `master` branch
- [ ] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

**Comment by kzarzycki** [comment]
It's 2025 and this missing feature is still live! Why the proposed PRs were closed?

This is a very needed feature now, because with closes the development loop for AI, allowing it to debug issues of strategies when running cloud backtests. Please prioritize this feature for implementation. At least reopen the ticket to gather votes!

**Comment by jaredbroad** [comment]
@kzarzycki the python tool box where the PR was made was deprecated and we made the LEAN CLI instead. The cloud data cannot be exported so providing a programmatic way of doing so via logs is too problematic for our data licenses today. We have to err on the side of caution to continue offering the data free via the cloud for the majority of users. 

We will extend the MCP tools to allow this cloud-cloud (cloud chat bot -> cloud backtest) and local-local (local LEAN CLI->any mcp caller). We don't have a local/LEAN-CLI MCP yet but will aim for it in 2026. 

We just deployed Mia V2 - an MCP enabled agentic copilot in the cloud IDE which is infinitely easier to use than manual MCP setups. It subsidizes token costs and once exceed subsidy passes through the token cost to clients.
